app/lib/smooch_nlu.rb

```diff
@@ -0,0 +1,115 @@
+class SmoochNlu
+  class SmoochBotNotInstalledError < ::ArgumentError
+  end
+
+  # FIXME: Make it more flexible
+  # FIXME: Once we support paraphrase-multilingual-mpnet-base-v2 make it the only model used
+  ALEGRE_MODELS_AND_THRESHOLDS = {
+    Bot::Alegre::OPENAI_ADA_MODEL => 0.8,
+    Bot::Alegre::MEAN_TOKENS_MODEL => 0.6
+  }
+
+  ALEGRE_CONTEXT_KEY = 'smooch_nlu_menu'
+
+  def initialize(team_slug)
+    @team_slug = team_slug
+    @smooch_bot_installation = TeamBotInstallation.where(team: Team.find_by_slug(team_slug), user: BotUser.smooch_user).last
+    raise SmoochBotNotInstalledError, "Smooch Bot not installed for workspace with slug #{team_slug}" if @smooch_bot_installation.nil?
+  end
+
+  def enable!
+    toggle!(true)
+  end
+
+  def disable!
+    toggle!(false)
+  end
+
+  def enabled?
+    !!@smooch_bot_installation.get_nlu_menus_enabled
+  end
+
+  def add_keyword(language, menu, menu_option_index, keyword)
+    update_keywords(language, menu, menu_option_index, keyword, 'add')
+  end
+
+  def remove_keyword(language, menu, menu_option_index, keyword)
+    update_keywords(language, menu, menu_option_index, keyword, 'remove')
+  end
+
+  def self.menu_option_from_message(message, options)
+    # FIXME: Raise exception if not in a tipline context (so, if Bot::Smooch.config is nil)
+    option = nil
+    if Bot::Smooch.config.to_h['nlu_menus_enabled']
+      # FIXME: No need to call Alegre if it's an exact match to one of the keywords
+      # FIXME: No need to call Alegre if message has no word characters
+      # FIXME: Handle error responses from Alegre
+      params = {
+        text: message,
+        models: ALEGRE_MODELS_AND_THRESHOLDS.keys,
+        per_model_threshold: ALEGRE_MODELS_AND_THRESHOLDS,
+        context: {
+          context: ALEGRE_CONTEXT_KEY,
+          team: Team.find(Bot::Smooch.config['team_id']).slug
```

Comment:
Can we just use `@team_slug` here since it is a class variable?
(Reading further I think `menu_option_from_message` is treated as a static / class function; so, it won't have this class variable)

Reply:
Yeah, not available at that scope... it's an instance variable while this method is a class method.